homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -355,7 +355,7 @@ def _get_home(hapid: str) -> Optional[AsyncHome]:
 
     # Register on HA stop event to gracefully shutdown HomematicIP Cloud connection
     hap.reset_connection_listener = hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STOP, async_reset_hap_connection()
+        EVENT_HOMEASSISTANT_STOP, await async_reset_hap_connection()
```

Comment:
This will be scheduled by `hass.async_add_job` by the event bus.

If we await here it will run the event handler here.

Reply:
To avoid the warning, make the event handler a callback that schedules the reset coroutine with `hass.async_create_task`.

Reply:
I am still doing something wrong.
Now reset_hap_connection is immediately called within async_setup_entry.

Reply:
Thanks @MartinHjelmare, works now.